openebs/pkg/volume/v1alpha1/volume.go

```diff
@@ -150,6 +150,9 @@ func (v CASVolume) ReadVolume(vname string, namespace string, obj interface{}) e
 	}
 
 	req.Header.Set("namespace", namespace)
+	// passing storageclass info as a request header which will extracted by the
+	// Maya-apiserver to get the CAS template name
+	req.Header.Set(string(v1alpha1.StorageClassKey), storageclass)
```

Comment:
There has been a change in maya api server w.r.t storage class key. Please refer to PR https://github.com/openebs/maya/pull/414